src/game-config.ts

```diff
@@ -1,5 +1,47 @@
-async function gameConfig() {
-  console.log("Not implemented yet!");
+import { main } from "./MainLoop";
+import { CommandLineArgumentParser } from "./util/ArgsParser";
+
+export interface GameConfig {
+  seed?: number;
+  // bots
+  kingdom?: string;
+
+  maxExpansions?: number;
+  disableExpansions?: Array<string>;
+  forceExpansions?: Array<string>;
+
+  disableCards?: Array<string>;
+  forceCards?: Array<string>;
+
+  useColonyPlat?: boolean;
+  cardOfTheDay?: boolean;
+  disableUi?: boolean;
+
+  debug?: boolean;
+}
+
+function gameConfig(): GameConfig {
+  const argParser = new CommandLineArgumentParser();
+  const args = argParser.parse();
+  console.log(args);
+
+  return {
+    seed: args.get("seed") as number | undefined,
+    // TODO: bot config
+    kingdom: args.get("kingdom") as string | undefined,
+
+    maxExpansions: args.get("max-expansions") as number | undefined,
+    disableExpansions: args.get("disable-expansions") as Array<string> | undefined,
+    forceExpansions: args.get("force-expansions") as Array<string> | undefined,
+
+    disableCards: args.get("disable-cards") as Array<string> | undefined,
+    forceCards: args.get("force-cards") as Array<string> | undefined,
```

Comment:
Better validation is required here. If the comma is missing it will be treated as a string and cause errors later on, so have some way to force this to be an array